lib/methods/action.ts

```diff
@@ -0,0 +1,94 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { sleep } from './utils'
+
+const DEFAULT_MAX_RETRIES = 30
+const DEFAULT_INITIAL_DELAY_MS = 1000
+const DEFAULT_RETRY_INTERVAL_MS = 2000
+
+/** Action is an interface that has a sys.status to be checked against */
+interface Action extends Record<string, any> {
+  sys: {
+    status: string
+  }
+}
+
+export class ActionProcessingError extends Error {
+  public action?: Action
+
+  constructor(message: string, action?: Action) {
+    super(message)
+    this.action = action
+    this.name = this.constructor.name
+  }
+}
+
+export class ActionFailedError extends ActionProcessingError {}
```

Comment:
TLDR here is that since Release Actions and BulkActions share a bit of the logic around failed/succeded actions I refactored it to accept anything that has a. `sys.status` property we can check.